st2common/st2common/services/executions.py

```diff
@@ -132,9 +142,12 @@ def _get_parent_execution(child_liveaction_db):
 def update_execution(liveaction_db, publish=True):
     execution = ActionExecution.get(liveaction__id=str(liveaction_db.id))
     decomposed = _decompose_liveaction(liveaction_db)
+    kw = {}
     for k, v in six.iteritems(decomposed):
-        setattr(execution, k, v)
-    execution = ActionExecution.add_or_update(execution, publish=publish)
+        kw['set__' + k] = v
```

Comment:
Surprisingly, `update` routine varies from `add_or_update` in most unexpected ways. It didn't work for me when I tried it initially and I've decided not to waste my time figuring out why.